python/pyarrow/includes/libarrow.pxd

```diff
@@ -1840,6 +1842,36 @@ cdef extern from "arrow/compute/api.h" namespace "arrow::compute" nogil:
         int num_args
         c_bool is_varargs
 
+        CArity()
+        CArity(int num_args, c_bool is_varargs)
+
+        @staticmethod
+        CArity Nullary()
+
+        @staticmethod
+        CArity Unary()
+
+        @staticmethod
+        CArity Binary()
+
+        @staticmethod
+        CArity Ternary()
+
+        @staticmethod
+        CArity VarArgs(int min_args)
```

Comment:
We don't need all of these overloads do we?

Reply:
Yes you’re right. We settled for the constructor.